extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClients.java

```diff
@@ -326,9 +328,41 @@ private MongoClientSettings createMongoConfiguration(MongoClientConfig config) {
             settings.uuidRepresentation(config.uuidRepresentation.get());
         }
 
+        settings = customize(name, settings);
+
         return settings.build();
     }
 
+    private boolean doesNotHaveClientNameQualifier(Bean<?> bean) {
+        for (Annotation qualifier : bean.getQualifiers()) {
+            if (qualifier.annotationType().equals(MongoClientName.class)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private MongoClientSettings.Builder customize(String name, MongoClientSettings.Builder settings) {
+        // If there is no customizer for the default client, use the first one that does not have a client name qualifier
+        // Otherwise look for the right customizer
```

Comment:
This comment seems not accurate, we use the first one that does not have a client name qualifier, if not we don't apply any customizer. The second line seems to be misleading.

Reply:
One day, I will explain you that comment... It is going to crush your deepest dreams. 

I fixed it.